### PR TITLE
tree: Rename more flex tree types to include "flex"

### PR DESCRIPTION
**Pull request by CraigMacomber**
## Description

Rename more of the internal flex tree APIs to use "flex" in their name to reduce conflicts/confusion with the simple-tree ones.

## Breaking Changes

Users of the internal flex tree APIs may need to rename some imports to include "flex"

## Reviewer Guidance

The review process is outlined on [this wiki page](https://github.com/microsoft/FluidFramework/wiki/PR-Guidelines#guidelines).

### CODE REVIEW

File: packages/dds/tree/src/index.ts [hunk 1]
```diff
@@ -148,7 +148,7 @@ export {
 	SequenceFieldEditBuilder,
 	prefixPath,
 	prefixFieldPath,
-	cursorForJsonableTreeNode as singleTextCursor,
+	cursorForJsonableTreeNode,
```
Comment (yann-achard-MS) [inline packages/dds/tree/src/index.ts:151]: This seems unrelated to this PR

**Review by taylorsw04 — APPROVED** [review]

File: packages/dds/tree/src/index.ts (hunk 1 above)
Comment (CraigMacomber) [reply packages/dds/tree/src/index.ts:151]: I purged the previously accidental `as` renames in the top level exports, this was one of them. I basically removed or baked in all the renames, so this is kinda unrelated, but kinda related.
